### PR TITLE
fix(common): make common/web/types kvk tests pass 🙀

### LINKED ISSUE #7967: chore(common): ldml some tests are failing  🙀
```shell
$ cd common/web/types
$ ./build.sh build
$ ./build.sh test
…
failures:

  ldml keyboard xml reader tests
    1) should fail to load files that don't conform to DTD
    2) should fail to load files with an invalid conformsTo
```

looks like these aren't run by CI?

#7042

### PR DESCRIPTION
**Pull request by srl295**
- use hexy and git-diff to show a rich diff when files are changed
- add additional round trip tests
- initialize the underlyingLayout to undefined to promote round tripping
- fix an issue with the CompilerCallbacks in the xml tests (broken in #8107)
- fix an issue with the kvk file writer where it should set strings irregardless (#7955)

Fixes #7967, again

Learning a bit about kvk…

@keymanapp-test-bot skip

### CODE REVIEW

File: common/web/types/src/kvk/kvk-file-writer.ts [hunk 1]
```diff
@@ -50,9 +50,7 @@ export default class KvkFileWriter {
         text: { len: 0, str: '' },
         bitmap: 0
       };
-      if (sourceKey.text) {
-        this.setString(binaryKey.text, sourceKey.text);
-      }
+      this.setString(binaryKey.text, sourceKey.text || '');
```
Comment (srl295) [inline common/web/types/src/kvk/kvk-file-writer.ts:53]: I introduced in #7955 but didn't realize that it broke common/web/types tests.

File: common/web/types/src/kvk/visual-keyboard.ts [hunk 2]
```diff
@@ -7,7 +7,7 @@
 import { BUILDER_KVK_SHIFT_STATE, BUILDER_KVK_HEADER_FLAGS, BUILDER_KVK_KEY_FLAGS } from "./kvk-file";
 
 export class VisualKeyboard {
-  header: VisualKeyboardHeader = {flags: 0, ansiFont:{}, unicodeFont:{}};
+  header: VisualKeyboardHeader = {flags: 0, ansiFont:{}, unicodeFont:{}, underlyingLayout: undefined};
```
Comment (srl295) [inline common/web/types/src/kvk/visual-keyboard.ts:10]: seems like noise, but it matches the kvks reader's parse() function

**Comment by srl295** [comment]
diff output something like this:

```
  2) kvks-file-reader
       kvks-file-reader should compile with kvk-file-writer:
     AssertionError: Buffers did not match:
@@ -4,138 +4,155 @@
 00000030: 0000 00f4 ffff ff08 0000 ff10 004b 0068  .............K.h
 00000040: 006d 0065 0072 0020 0042 0075 0073 0072  .m.e.r. .B.u.s.r
 00000050: 0061 0020 004b 0062 0064 0000 0010 0000  .a. .K.b.d......
-00000060: 0008 0000 ffba 0000 0002 4000 4200 0200  ..........@.B...
-00000070: 9e17 0000 0000 0000 0240 004b 0002 009d  .........@.K....
-00000080: 1700 0000 0000 0002 4000 de00 0200 c817  ........@.......
-00000090: 0000 0000 0000 0240 00dd 0002 00b3 1700  .......@........
-000000a0: 0000 0000 0002 4000 5400 0200 a817 0000  ......@.T.......
-000000b0: 0000 0000 0240 00db 0002 00a9 1700 0000  .....@..........
-000000c0: 0000 0002 4000 5000 0200 b017 0000 0000  ....@.P.........
```

could move to a common area perhaps?

**Review by mcdurdin — APPROVED** [review]
LGTM

**Pull request merged**